src/Util/Config/ConfigFileSaver.php

```diff
@@ -0,0 +1,341 @@
+<?php
+
+namespace Friendica\Util\Config;
+
+/**
+ * The ConfigFileSaver saves specific variables into the config-files
+ *
+ * It is capable of loading the following config files:
+ * - *.config.php   (current)
+ * - *.ini.php      (deprecated)
+ * - *.htconfig.php (deprecated)
+ */
+class ConfigFileSaver extends ConfigFileManager
+{
+	/**
+	 * The standard indentation for config files
+	 * @var string
+	 */
+	const INDENT = "\t";
+
+	/**
+	 * The settings array to save to
+	 * @var array
+	 */
+	private $settings = [];
+
+	/**
+	 * Adds a given value to the config file
+	 * Either it replaces the current value or it will get added
+	 *
+	 * @param string $cat   The configuration category
+	 * @param string $key   The configuration key
+	 * @param string $value The new value
+	 */
+	public function addConfigValue($cat, $key, $value)
+	{
+		$settingsCount = count(array_keys($this->settings));
+
+		for ($i = 0; $i < $settingsCount; $i++) {
+			// if already set, overwrite the value
+			if ($this->settings[$i]['cat'] === $cat &&
+				$this->settings[$i]['key'] === $key) {
+				$this->settings[$i] = ['cat' => $cat, 'key' => $key, 'value' => $value];
+				return;
+			}
+		}
+
+		$this->settings[] = ['cat' => $cat, 'key' => $key, 'value' => $value];
+	}
+
+	/**
+	 * Resetting all added configuration entries so far
+	 */
+	public function reset()
+	{
+		$this->settings = [];
+	}
+
+	/**
+	 * Save all added configuration entries to the given config files
+	 * After updating the config entries, all configuration entries will be reseted
+	 *
+	 * @param string $name The name of the configuration file (default is empty, which means the default name each type)
+	 *
+	 * @return bool true, if at least one configuration file was successfully updated or nothing to do
+	 */
+	public function saveToConfigFile($name = '')
```

Comment:
Correct me if I'm wrong, but isn't this method able to automatically convert any install config files to the latest format?

You see where I'm going with this, right?

Reply:
Ha, yes that's possible :-)

So we create an automatic upgrade path for legacy files. That sounds exciting